server/test/unit/server/webservices/views/test_util.py

```diff
@@ -165,3 +165,27 @@ def mock_function(self, request):
 
         self.assertEqual(response.http_status_code, 400)
         self.assertEqual(response.error_code.code, 'PLP1009')
+
+
+class TestEnsureInputEncoding(unittest.TestCase):
+
+    def test_ensure_invalid_input_encoding(self):
+        """
+        Test invalid input encoding
+        """
+        input_data = {"invalid": "json\x81"}
+        try:
+            util._ensure_input_encoding(input_data)
+        except InputEncodingError:
```

Comment:
Unless you want to save the exception in this line, you can use a `self.assertRaises` instead. The pattern that does not work in older versions of python is:

    with self.asserRaises(blah):
        do_stuff()